service/src/main/java/org/databiosphere/workspacedataservice/dao/PostgresJobDao.java

```diff
@@ -235,43 +249,87 @@ public List<GenericJobServerModel> getJobsForCollection(
+      var updated = rs.getTimestamp("updated").toLocalDateTime().atOffset(ZoneOffset.UTC);
+      UUID collectionId = rs.getObject("collection_id", UUID.class);
+
+      // TODO: AJ-1011 also return stacktrace, result.
+      return new GenericJobServerModel(
+              jobId, jobType, /* instanceId= */ collectionId, status, created, updated)
+          .errorMessage(rs.getString("error"))
+          .input(getJobInput(jobType, rs));
+    }
 
-      JobTypeEnum jobType = JobTypeEnum.UNKNOWN;
+    private JobTypeEnum getJobType(ResultSet rs) throws SQLException {
       String jobTypeStr = rs.getString("type");
       try {
-        jobType = JobTypeEnum.fromValue(jobTypeStr);
+        return JobTypeEnum.fromValue(jobTypeStr);
       } catch (IllegalArgumentException ill) {
         logger.warn("Unexpected JobTypeEnum found: {}", jobTypeStr);
       }
+      return JobTypeEnum.UNKNOWN;
+    }
 
-      StatusEnum status = StatusEnum.UNKNOWN;
+    private StatusEnum getStatus(ResultSet rs) throws SQLException {
       String statusStr = rs.getString("status");
       try {
-        status = StatusEnum.fromValue(statusStr);
+        return StatusEnum.fromValue(statusStr);
       } catch (IllegalArgumentException ill) {
         logger.warn("Unexpected StatusEnum found: {}", statusStr);
       }
+      return StatusEnum.UNKNOWN;
+    }
 
-      var created = rs.getTimestamp("created").toLocalDateTime().atOffset(ZoneOffset.UTC);
-      var updated = rs.getTimestamp("updated").toLocalDateTime().atOffset(ZoneOffset.UTC);
-
-      UUID collectionId = rs.getObject("collection_id", UUID.class);
-      GenericJobServerModel job =
-          new GenericJobServerModel(jobId, jobType, collectionId, status, created, updated);
+    private JobInput getJobInput(JobTypeEnum jobType, ResultSet rs) throws SQLException {
+      Class<? extends JobInput> targetClass =
+          DATA_IMPORT.equals(jobType) ? ImportJobInput.class : JobInput.class;
+      try {
+        return mapper.readValue(rs.getString("input"), targetClass);
+      } catch (JsonProcessingException e) {
+        logger.error("Error deserializing input: {}; input will be empty.", e.getMessage());
+      }
+      return JobInput.empty();
+    }
+  }
 
-      job.errorMessage(rs.getString("error"));
-      job.instanceId(rs.getObject("collection_id", UUID.class));
+  private GenericJobServerModel measureElapsedTime(GenericJobServerModel job, StatusEnum status) {
+    metrics
+        .timer("wds.job.elapsed", getTags(job, status))
+        .record(
+            Duration.between(
+                job.getCreated(), Instant.ofEpochMilli(clock.wallTime()).atOffset(ZoneOffset.UTC)));
+    return job;
+  }
 
-      // TODO: AJ-1011 also return stacktrace, input, result.
-      return job;
+  private static Tags getTags(GenericJobServerModel job, StatusEnum status) {
+    Tags tags = Tags.of("jobType", job.getJobType().toString());
+    tags = tags.and("newStatus", status.toString());
```

Comment:
Can these two lines be combined?

```suggestion
    Tags tags = Tags.of("jobType", job.getJobType().toString(), "newStatus", status.toString());
```

Reply:
Yes they can be, but I actively try to avoid `...varargs` invocations where you can lose track of key/value pairs.  I prefer to be explicit with the 2-arg constructors where it's clear which thing is key vs. value

This is why I always use `ImmutableMap.Builder` for building up many-valued `Map`s instead of `Map.of`